tests/test_dataset.py

```diff
@@ -545,6 +545,15 @@ def create_index(self):
         with pytest.raises(RuntimeError):
             _ = TestDataset().index
 
+    def test_warning_index_no_valid_attribute(self):
```

Comment:
Just for completness, let's test for a couple of different namings (i.e. with `-`, some other special character, Pyhton builtin name...).

You can use `pytest.mark.paramerterize` for this